src/HLSMonitorService.ts

```diff
@@ -92,43 +166,43 @@ export class HLSMonitorService {
         .send({ status: "Healthy 💖" });
     });
 
-    this.fastify.get("/stop", async (request, reply) => {
-      if (!this.hlsMonitor) {
+    this.fastify.post("/monitor/:monitorId/stop", async (request, reply) => {
+      if (!this.hlsMonitors.has(request.params.monitorId)) {
         reply.code(500).send({
           status: "error",
           message: "monitor not initialized",
         });
       }
-      await this.hlsMonitor.stop();
+      await this.hlsMonitors.get(request.params.monitorId).stop();
       reply
         .code(200)
         .header('Content-Type', 'application/json; charset=utf-8')
         .send({ status: "Stopped monitoring" });
     });
 
-    this.fastify.get("/start", async (request, reply) => {
-      if (!this.hlsMonitor) {
+    this.fastify.post("/monitor/:monitorId/start", async (request, reply) => {
+      if (!this.hlsMonitors.has(request.params.monitorId)) {
         reply.code(500).send({
           status: "error",
           message: "monitor not initialized",
         });
       }
-      this.hlsMonitor.start();
+      this.hlsMonitors.get(request.params.monitorId).start();
       reply
         .code(200)
         .header('Content-Type', 'application/json; charset=utf-8')
         .send({ status: "Started monitoring" });
     });
 
-    this.fastify.put("/delete", async (request, reply) => {
-      if (!this.hlsMonitor) {
+    this.fastify.put("/monitor/:monitorId/delete", async (request, reply) => {
```

Comment:
I suggest using HTTP method DELETE on this resource instead, i.e.

`HTTP DELETE /monitor/:monitorId`